### PR TITLE
Patch incorrect image scaling in carousel slide in MS Edge

### PR DESCRIPTION
**Pull request by sanusart**
Fix correct image scaling in carousel slide in Microsoft Edge browser - issue #16915

### CODE REVIEW

**Comment by kkirsche** [comment]
Please post a JSBin of the change since this would need extensive cross browser testing since this is only a hug in Edge (which Microsoft devs are looking into per the response to @cvrebert's tweet)

**Comment by sanusart** [comment]
Off course. Hope it'll help debug and test:

# [JSBin](http://jsbin.com/wabazavuya/edit?html,output)

### Microsoft Edge (w/o fix):
image

### Same JSBin in Google chrome:
image

### Same JSBin in Mozilla Firefox
image

# [JSBin (with fix)](http://jsbin.com/semehodejo/1/edit?html,output)

### Microsoft Edge (with fix):
image

### Google chrome (with fix):
image

### Mozilla Firefox with (with fix):
image

File: less/mixins/image.less [hunk 1]
```diff
@@ -8,6 +8,7 @@
 // Keep images from scaling beyond the width of their parents.
 .img-responsive(@display: block) {
   display: @display;
+  width: 100%; // Allow correct scaling of image on carousel slide in MS Edge https://github.com/twbs/bootstrap/issues/16915
```
Comment (cvrebert) [inline less/mixins/image.less]: This makes the `max-width` on the next line redundant.

**Comment by cvrebert** [comment]
I presume the code currently uses `max-width` instead of `width` to avoid any stretching of the image. The difference should be noticeable if you use a relatively small image instead of Holder.js.

**Comment by sanusart** [comment]
I added and not replaced because basically, `max-width` will override `width` (to be on the safe side of it), as the desirable outcome is to have image stretched no matter of it's original size I presume. 
I'll test with smaller images tomorrow and post the result.

**Comment by sanusart** [comment]
# Test with small images

Images stretched to thier full width indeed while `width: 100%` applied. But isn't it the desired behavior in such component as carousel?

# [JSBin](http://jsbin.com/jadaqorohi/edit?html,output)

### WIndows Edge

image

### Google Chrome

image

### Mozilla Firefox

image

**Comment by cvrebert** [comment]
CC: @mdo

**Comment by mdo** [comment]
What about IE 8-11?

**Comment by cvrebert** [comment]
IE8:
ie8

IE9:
ie9

IE10:
ie10

IE11:
ie11

**Comment by mdo** [comment]
Rad, seems fine to me then, eh?

**Comment by sanusart** [comment]
I think IE8-11 should be tested with `html5shim` and `respond.js`: 
[JSBin](http://jsbin.com/bujozuleju/1/edit?html,output)

As this is the [suggested](http://getbootstrap.com/getting-started/#template) template for IE=<8

**Comment by cvrebert** [comment]
html5shiv and Respond.js shouldn't affect the results in this case AFAIK. The example doesn't use HTML5 elements or media queries.

**Comment by cvrebert** [comment]
> Rad, seems fine to me then, eh?

I guess. It just feels a little wrong to me to change this just to satisfy Edge, considering that @jonathansampson [said the Edge dev team would look into this](https://twitter.com/jonathansampson/status/626849106430922752) on their end. But pragmatism, I suppose.

**Comment by sanusart** [comment]
Yep. Let's see what happens with the Edge team.

**Comment by mdo** [comment]
Did you ever get an update on this @cvrebert?

**Comment by mdo** [comment]
Wait, shouldn't this only be applied to the carousel items, and not the entire `img-responsive` mixin?

**Comment by sanusart** [comment]
It probably should in some cases, but the up side of this, is that it can
serve other "Edge" cases in which image with class `img-responsive` is
wrapped in block element.
Do you see any harm in this way?

**Comment by cvrebert** [comment]
> Did you ever get an update on this @cvrebert?

Not really. @jonathansampson agreed that the `width: 100%` fix seemed valid, but never gave an update as to (dis)confirming it as a bug in Edge.

**Comment by mdo** [comment]
Making a change like this so late in v3's development doesn't sit right with me, so I'm super hesitant to accept it right now. We could explore it in v4 with thorough cross browser testing, but if there's a huge bug here for Edge with the carousel, let's just fix that.

**Comment by sanusart** [comment]
[Test jsbin](http://jsbin.com/titirufanu/1/edit?css,output) with compiled less:
screenshot_2015-11-16_23_07_31

**Comment by cvrebert** [comment]
> Did you ever get an update on this @cvrebert?

Came up with a clean testcase and filed an Edge bug:
https://connect.microsoft.com/IE/feedback/details/2274240/edge-edge-renders-an-svg-img-with-max-width-100-too-small

**Comment by cvrebert** [comment]
The relevant bug is no longer present as of the latest stable version of Edge (25.10586.0.0). :smile: 
Thus, I'm closing this PR since its changes should no longer be necessary.
